src/librustdoc/html/render/mod.rs

```diff
@@ -1172,21 +1181,19 @@ fn render_assoc_items(
             AssocItemRender::DerefFor { trait_, type_, deref_mut_ } => {
                 let id = cx.derive_id(small_url_encode(format!(
                     "deref-methods-{:#}",
-                    type_.print(cx.cache())
+                    type_.print(cache, tcx)
```

Comment:
Are all these changes from `cx.cache()` to `cache` necessary? They're not *bad*, but they make an already giant diff even bigger, it would be nice to put them in a separate commit or something.